scripts/whatsdue.js

```diff
@@ -97,9 +97,14 @@ module.exports = function (robot) {
 
     /**
      * Robot responds to a message containing `!whatsdue`.
-     * Requires at least one course.
      */
-    robot.respond(/!?whatsdue ((?:[a-z]{4}[0-9]{4} ?)+)/i, function (res) {
+    robot.respond(/!?whatsdue ?((?:[a-z]{4}[0-9]{4} ?)+)?/i, function (res) {
+        // If there are no matching groups, throw an error.
+        if (!res.match[1]) {
```

Comment:
happy to change it if there is another compelling reason to :) but atm I think it reads better than if(res.match.length != 2) or whatever hehe

appreciate the comment Tom!